### PR TITLE
fix: feeType=positive_slippage logic [TKR-81]

### PR DESCRIPTION
**Pull request by romainbutteaud**
# Description

- API change: https://github.com/0xProject/protocol/pull/209

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/utils/service_utils.ts [hunk 1]
```diff
@@ -106,7 +106,12 @@ export const serviceUtils = {
         }, []);
     },
     getAffiliateFeeAmounts(quote: SwapQuote, fee: AffiliateFee): AffiliateFeeAmounts {
-        if (fee.feeType === AffiliateFeeType.None || fee.recipient === NULL_ADDRESS || fee.recipient === '') {
+        if (
+            fee.feeType === AffiliateFeeType.None ||
+            fee.recipient === NULL_ADDRESS ||
+            fee.recipient === '' ||
+            (fee.feeType === AffiliateFeeType.PositiveSlippageFee && quote.orders.length === 1)
```
Comment (dekz) [inline src/utils/service_utils.ts:113]: What's the intention here where quote `orders.length === 1`

Comment (romainbutteaud) [reply src/utils/service_utils.ts:113]: I don't want to add `affiliateFee` gas overhead if the quote has `PositiveSlippageFee` and if we're only routing to 1 order (VIP). Just kind of like here: https://github.com/0xProject/protocol/blob/6d877d52425230ba7664a0685ab0a06957108acb/packages/asset-swapper/src/quote_consumers/quote_consumer_utils.ts#L82

Comment (romainbutteaud) [reply src/utils/service_utils.ts:113]: What do you think?

**Comment by kimpers** [comment]
We doing anything with this @romainbutteaud, @dekz?

**Pull request closed without merging**